pkg/flash/flash_linux.go

```diff
@@ -54,16 +60,36 @@ func New(spi SPI) (*Flash, error) {
 
 	var err error
 	f.sfdp, err = sfdp.Read(f.SFDPReader())
-	if err != nil {
-		return nil, fmt.Errorf("could not read sfdp: %v", err)
+	if err == nil {
+		if err = f.FillFromSFDP(); err == nil {
+			return f, nil
+		}
 	}
 
+	id, e := f.spi.ID()
+	if e != nil {
+		return nil, errors.Join(err, e)
+	}
+
+	if f.chip, e = chips.New(id); e != nil {
+		return nil, errors.Join(err, e)
+	}
+
+	f.is4ba = f.chip.Is4BA
```

Comment:
I think it's enough to just store the chip in f.chip. No need to have copies of all the fields.